### PR TITLE
72 add query helper utilities

### PR DESCRIPTION
**Pull request by AD1306**
* Added a new class `QueryBuilder` to help out with making queries.

### CODE REVIEW

**Comment by tanandre** [comment]
You are not encoding the URL parameters? I'm missing use of `encodeURIComponent`

File: src/library/QueryBuilder/QueryBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  /**
+   * @returns complete generated query with decoded parameter values
+   */
+  getCompleteUrlDecoded(): String {
+    return decodeURIComponent(this.getCompleteUrl().toString()); 
+  }
+  
+  private createSortQuery(): String {
+    let sortQuery = "";
+    const sortDelimeter = this.singularQueries.length > 0 ? this.QUERY_DELIMETER : "";
```
Comment (tanandre) [inline src/library/QueryBuilder/QueryBuilder.ts:111]: I don't like that this createSortQuery() function has to know about the field singularQueries. Refactoring will make this reference hard to trace. Either pass it in as a parameter or, better yet, put the logic for the delimiter in the calling function (getCompleteUrl()).

Comment (AD1306) [reply src/library/QueryBuilder/QueryBuilder.ts]: i actually passed it in as a parameter just to keep the code a bit tidy.

**Comment by tanandre** [comment]
looks good

**Review by tanandre — APPROVED** [review]
looks good

**Pull request merged**